### DISCUSSION TITLE
Adicionar links de redes sociais aos perfis

### DISCUSSION THREAD

**Discussion by malikpiara** [outono-org/.github#3] [Pitch]
### Problema

> Que dor é que estamos a tentar aliviar ou remover?

Parte do objetivo da Outono é ajudar pessoas a encontrar outras pessoas com interesses semelhantes. Mas neste momento, não há forma de conectar com outra pessoa ou começar uma conversa.

### Investimento

> Quanto tempo é que queremos investir neste problema?

1-2 dias é o tempo apropriado. Investir mais tempo seria um desperdício porque ainda não temos user insights suficientes para construir funcionalidades mais desenvolvidas.

### Solução que eu proponho

Adicionar ícones com links para outras redes sociais aos perfis dos utilizadores.
É uma solução sem risco e que faz sentido, independentemente da direção que a plataforma tome no futuro. O backend já foi desenvolvido, facilitando este esforço. 

![PXL_20230516_143012010](https://github.com/outono-org/.github/assets/6923650/bfbe0e2d-c40e-4531-8839-38e83b743623)

Na imagem acima, os ícones são adicionados no lugar do botão "website" que existe atualmente. O botão website tornar-se-ia também num ícone com um design consistente com os outros ícones.

### Alguns requisitos
1. Os links devem abrir noutro separador.
2. Só devemos mostrar ícones para os links das redes sociais que o utilizador adicionou.
3. Os ícones devem ter um design consistente.
4. Os ícones devem ter preferencialmente o formato .svg
5. Neste momento, o backend suporta as seguintes redes sociais: LinkedIn, GitHub, Dribbble, Instagram, Twitter. Temos de nos certificar que temos ícones para todas.
6. Devemos considerar acessibilidade. Todos os ícones devem incluir labels e aria tags.

**Comment by malikpiara** [comment]
@Neuville Acho que este improvement é um bom starting point para poderes começar a contribuir. Se houver alguma questão em relação a este pitch ou se precisares de apoio, let me know.

**Reply by Neuville** [reply]
@malikpiara  Criei o primeiro test com apenas com os svgs sem links. Diz-me só quais são os fields que a tabela usa para ir buscar os links
![Captura de ecrã 2023-05-17, às 15 26 11](https://github.com/outono-org/.github/assets/36772385/2f4c12dc-0078-48c8-a65c-8fb03064733b)

**Reply by malikpiara** [reply]
> @malikpiara Criei o primeiro test com apenas com os svgs sem links. Diz-me só quais são os fields que a tabela usa para ir buscar os links ![Captura de ecrã 2023-05-17, às 15 26 11](https://user-images.githubusercontent.com/36772385/238982183-2f4c12dc-0078-48c8-a65c-8fb03064733b.png)

Hey @Neuville, altamente! Tens de trabalhar no branch #social-media. Criei uma helper function que faz fetch dos links das redes sociais. Podes utilizá-la:
https://github.com/outono-org/outono.org/blob/social-links/utils/getOnlineProfiles.js

Se te ajudar a visualizar, tens aqui um screenshot da tabela _user_online_profiles_:

<img width="934" alt="image" src="https://github.com/outono-org/.github/assets/6923650/a50812f7-3a2a-4038-9d36-c1e1991d8124">

**Reply by Neuville** [reply]
> ####
> Comentário original em português - Traduzir para português

Hey @malikpiara, já pus lá as mudanças e mais logo uso a função para chamar os links 👍

**Reply by Neuville** [reply]
@malikpiara a feature está a funcionar!
![itworks](https://github.com/outono-org/.github/assets/36772385/9fc3aba8-77cf-4c4d-8b10-1a85a0cfa6a5)

**Reply by malikpiara** [reply]
Já dei pull e estive a testar no meu computador. Acho que tens de dar uma vista de olhos à descrição e aos requisitos mas estamos num bom caminho. Quando estiveres pronto para um code review antes de dar merge apita 🙌
